crates/subspace-node/src/service.rs

```diff
@@ -307,16 +307,15 @@ pub async fn new_full(config: Configuration) -> Result<NewFull<Arc<FullClient>>,
         );
     }
 
-    let role = config.role.clone();
-    let force_authoring = config.force_authoring;
     let backoff_authoring_blocks: Option<()> = None;
     let prometheus_registry = config.prometheus_registry().cloned();
 
     let new_slot_notification_stream = subspace_link.new_slot_notification_stream();
     let block_signing_notification_stream = subspace_link.block_signing_notification_stream();
     let archived_segment_notification_stream = subspace_link.archived_segment_notification_stream();
 
-    let overseer_handle = if let Some(_keystore) = keystore_container.local_keystore() {
+    // TODO: In the future we want `--executor` CLI flag instead
+    let overseer_handle = if config.role.is_authority() {
```

Comment:
Having `is_authority()` as true does not mean that the overseer should be enabled, the overseer only makes sense for the embedded node, not the regular node. I think the right way is to add a new flag `--executor` and use `is_executor()` or something like that.

Reply:
Does it hurt anything if it is enabled though? If not I think it makes sense to just have TODO and worry about it later as this code will change when we embed executor into this node anyway.

Reply:
Well, no harm at all, just spawned an extra unnecessary task. I'm fine with leaving it a TODO.

Reply:
If you want feel free to update this PR to include new CLI flag